Orleans.Providers.MongoDB/Reminders/Store/MongoReminderCollection.cs

```diff
@@ -40,43 +40,37 @@ protected override void SetupCollection(IMongoCollection<MongoReminderDocument>
                     .Ascending(x => x.IsDeleted)
                     .Ascending(x => x.ServiceId)
                     .Ascending(x => x.GrainHash);
-            try
-            {
-                collection.Indexes.CreateOne(
-                    new CreateIndexModel<MongoReminderDocument>(byHashDefinition,
-                        new CreateIndexOptions
-                        {
-                            Name = "ByHash"
-                        }));
-            }
-            catch (MongoCommandException ex)
-            {
-                if (ex.CodeName == "IndexOptionsConflict")
-                {
-                    collection.Indexes.CreateOne(new CreateIndexModel<MongoReminderDocument>(byHashDefinition));
-                }
-            }
+
+            CreateIndex("ByHash", collection, byHashDefinition);
 
             var byNameDefinition =
                 Index
                     .Ascending(x => x.IsDeleted)
                     .Ascending(x => x.ServiceId)
                     .Ascending(x => x.GrainId)
                     .Ascending(x => x.ReminderName);
+
+            CreateIndex("ByName", collection, byNameDefinition);
+
+            var byGrainIdDefinition =
```

Comment:
I think the fields should probably be reversed here. It better aligns with filtering down the service's grains first then finding the specific grain within that service:
```csharp
Index
    .Ascending(x => x.ServiceId)
    .Ascending(x => x.GrainId);
```

And it will cover this (less used) query too:
```csharp
return Collection.DeleteManyAsync(r => r.ServiceId == serviceId);
```